src/Html.php

```diff
@@ -4970,110 +4971,41 @@ public static function jsAjaxDropdown($name, $field_id, $url, $params = [])
-            allowClear: $allowclear,
-            minimumInputLength: 0,
-            quietMillis: 100,
-            dropdownAutoWidth: true,
-            dropdownParent: $('#$field_id').closest('div.modal, div.dropdown-menu, body'),
-            minimumResultsForSearch: " . $CFG_GLPI['ajax_limit_count'] . ",
-            ajax: {
-               url: '$url',
-               dataType: 'json',
-               type: 'POST',
-               data: function (params) {
-                  query = params;
-                  return $.extend({}, params_$field_id, {
-                     searchText: params.term,";
-
-        if ($parent_id_field !== null) {
-            $js .= "
-                     parent_id : document.getElementById('" . $parent_id_field . "').value,";
-        }
-        $js .= "
-                     page_limit: " . $CFG_GLPI['dropdown_max'] . ", // page size
-                     page: params.page || 1, // page number
-                  });
-               },
-               processResults: function (data, params) {
-                  params.page = params.page || 1;
-                  var more = (data.count >= " . $CFG_GLPI['dropdown_max'] . ");
-
-                  return {
-                     results: data.results,
-                     pagination: {
-                           more: more
-                     }
-                  };
-               }
-            },
-            templateResult: templateResult,
-            templateSelection: templateSelection
-         })
-         .bind('setValue', function(e, value) {
-            $.ajax('$url', {
-               data: $.extend({}, params_$field_id, {
-                  _one_id: value,
-               }),
-               dataType: 'json',
-               type: 'POST',
-            }).done(function(data) {
-
-               var iterate_options = function(options, value) {
-                  var to_return = false;
-                  $.each(options, function(index, option) {
-                     if (option.hasOwnProperty('id')
-                         && option.id == value) {
-                        to_return = option;
-                        return false; // act as break;
-                     }
-
-                     if (option.hasOwnProperty('children')) {
-                        to_return = iterate_options(option.children, value);
-                     }
-                  });
-
-                  return to_return;
-               };
+        $js = <<<JS
+            select2_configs['{$field_id}'] = {
+                field_id: '{$field_id}',
+                width: '{$width}',
+                multiple: '{$multiple}',
+                placeholder: '{$placeholder}',
+                allowclear: {$allowclear},
+                ajax_limit_count: {$CFG_GLPI['ajax_limit_count']},
+                dropdown_max: {$CFG_GLPI['dropdown_max']},
+                url: '{$url}',
+                parent_id_field: '{$parent_id_field}',
+                on_change: '{$on_change}',
```

Comment:
`$on_change` may probably contains a single quote. It has to be either escaped, either encoded using `json_encode()`. `json_encode()` is safer but I do not know if it would work as expected.